agenta-cli/agenta/sdk/decorators/llm_entrypoint.py

```diff
@@ -68,12 +68,17 @@ async def chain_of_prompts_llm(prompt: str):
     ```
     """
 
-    def __call__(self, func: Callable[..., Any]) -> Callable[..., Any]:
+    def __call__(self, func: Callable[..., Any]):
         endpoint_name = "generate"
         func_signature = inspect.signature(func)
         config_params = agenta.config.all()
         ingestible_files = self.extract_ingestible_files(func_signature)
 
+        if getattr(func, "__wrapped__", None) is None:
```

Comment:
I don't think this is right. You are checking whether the `generate` function has already some other decorator then throwing an error if now. This makes two incorrect assumptions:
- That the user should be forced to use `@ag.trace`
- That the user might not use another decorator instead of trace instead of that.
Basically we are testing for a different thing (the existence of another decorator before `entrypoint`) then what we are throwing an error for (`@trace` comes after `@entrypoint`). 

I don't think this is the right solution

Reply:
Thank you for your review, @mmabrouk. You are correct that the current solution makes two incorrect assumptions. I have updated the solution to address your concerns. The solution I came up with allows the user to enable tracing using the `entrypoint` decorator. When enabled, it wraps the llm app function (`func`) with the `trace` decorator. I have updated the `tracing_plus_entrypoint` workflow example to show how the entrypoint will be used.